### ISSUE TITLE
onClick event doesn't work with the SelectionContainer

### ISSUE THREAD

**Issue by sarah-clements** [FormidableLabs/victory#1367]
# Bugs and Questions

### Checklist

- [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))

- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question

- [x] I am using the latest version of Victory

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### The Problem

The selection container should have onClick events enabled for selecting datapoints but it doesn't. Other events like onMouseOver works fine.

### Reproduction

See demo [here](https://codesandbox.io/s/selection-container-bug-3kpm0?fontsize=14)

**Comment by boygirl** [comment]
@sarah-clements thanks for the reproduction! It looks like I'm missing an `event.stopPropagation` on the `onMouseDown` event in `VictorySelectionContainer`. Let me add that and see if it helps

**Comment by sarah-clements** [comment]
> @sarah-clements thanks for the reproduction! It looks like I'm missing an `event.stopPropagation` on the `onMouseDown` event in `VictorySelectionContainer`. Let me add that and see if it helps

Thanks for looking into it. Do you know if this patch will be added to the next release? It's a big blocker for adopting this container.

**Comment by boygirl** [comment]
@sarah-clements sorry for the delay. The simple fix I anticipated did not solve the issue. But in my investigation, I noticed that while `onClick` is not working, `onMouseDown` is. So this might be a good temporary work around for you until I get this sorted.

**Comment by boygirl** [comment]
@sarah-clements okay, after just a bit more digging, I realized my initial idea was correct, but backwards. The events you register on `VictoryScatter` will need to include an `onMouseDown` event that stops propagation in order to keep the events on the container from interfering with the completion of the click event. So, adding the following to your `eventHandlers` object should do it:
```
onMouseDown: evt => evt.stopPropagation()
```
I've edited your example as well.
https://codesandbox.io/s/selection-container-bug-resolution-mtlri

I don't think there's a good way to fix this within Victory, but I will open an issue to document difficulties with click events and evented containers.
